RNTester/NativeModuleExample/Screenshot.mm

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#import "Screenshot.h"
+#import <React/RCTUIManager.h>
+#import <React/RCTUtils.h>
+#import <ReactCommon/RCTTurboModuleManager.h>
+#import <ReactCommon/TurboModuleUtils.h>
+
+static NSString* SaveScreenshotToTempFile()
+{
+   // find the key window
+  NSWindow* keyWindow;
+  for (NSWindow* window in NSApp.windows) {
+    if (window.keyWindow) {
+      keyWindow = window;
+      break;
+    }
+  }
+
+  // take a snapshot of the key window
+  CGWindowID windowID = (CGWindowID)[keyWindow windowNumber];
+  CGWindowImageOption imageOptions = kCGWindowImageDefault;
+  CGWindowListOption listOptions = kCGWindowListOptionIncludingWindow;
+  CGRect imageBounds = CGRectNull;
+  CGImageRef windowImage = CGWindowListCreateImage(
+    imageBounds,
+    listOptions,
+    windowID,
+    imageOptions);
+  NSImage* image = [[NSImage alloc] initWithCGImage:windowImage size:[keyWindow frame].size];
+
+  // save to a temp file
```

Comment:
> One thing that isn’t immediately clear to me is why this is using Turbo Modules; is it just forward thinking or did you take this module as an entry-point to start your larger work effort on Turbo Modules in RN macOS?

Yes. This is an existing NativeModule demo for iOS RNTester, so I think making it work in mac using turbo module is a good start. So the detail about taking a snapshot is not important, the important thing is that it is get called via turbo module and it is possible to make threading things work.

Later I will move some code from react-native-windows, so that to make turbo module being able to be implemented in pure C++, not by talking to JSI directly like this.

Reply:
Gotcha, thanks for elaborating 🙏

Reply:
The original comment in this thread still stands, though, how about not compressing/writing to disk on the main thread?